src/sentry/integrations/grpc/protos/scm.proto

```diff
@@ -0,0 +1,512 @@
+// RepoTreesIntegration Request/Response Messages
+// ============================================================================
+
+message GetTreesForOrgRequest {
+  int64 organization_id = 1;
+  Provider provider = 2;
+}
+
+message GetTreesForOrgResponse {
+  repeated RepoTree trees = 1;
+}
+
+message GetCachedRepoFilesRequest {
+  int64 organization_id = 1;
+  Provider provider = 2;
+  string repo_full_name = 3;
+  string tree_sha = 4;
+  int32 shifted_seconds = 5;
+  bool only_source_code_files = 6;
+  bool only_use_cache = 7;
+}
+
+message GetCachedRepoFilesResponse {
+  repeated string files = 1;
+}
+
+// ============================================================================
+// SourceCodeIssueIntegration Request/Response Messages
+// ============================================================================
+
+message GetRepositoryChoicesRequest {
+  int64 organization_id = 1;
+  Provider provider = 2;
+  int64 group_id = 3;
+  map<string, string> params = 4;
+  int32 page_size = 5;
+}
+
+message GetRepositoryChoicesResponse {
+  string default_repository_id = 1;
+  repeated RepositoryChoice choices = 2;
+}
+
+message GetProjectDefaultsRequest {
+  int64 organization_id = 1;
+  Provider provider = 2;
+  int64 project_id = 3;
+}
+
+message GetProjectDefaultsResponse {
+  map<string, string> defaults = 1;
+}
+
+// ============================================================================
+// Legacy/Backwards Compatibility (keeping some old messages for now)
+// ============================================================================
+
+message ListRepositoriesRequest {
+  int64 organization_id = 1;
+  Provider provider = 2;
+  string query = 3;
+  int32 page_size = 4;
+  string page_token = 5;
+}
+
+message ListRepositoriesResponse {
+  repeated Repository repositories = 1;
+  string next_page_token = 2;
+}
+
+message GetRepositoryRequest {
+  int64 repository_id = 1;
+}
+
+message GetCommitRequest {
+  int64 repository_id = 1;
+  string commit_sha = 2;
+}
+
+message GetCommitContextRequest {
+  int64 organization_id = 1;
+  int64 repository_id = 2;
+  string commit_sha = 3;
+  string filepath = 4;
+  int32 line_number = 5;
+  Provider provider = 6;
+}
+
+message CreateCodeMappingRequest {
+  int64 organization_id = 1;
+  int64 project_id = 2;
+  int64 repository_id = 3;
+  string stack_root = 4;
+  string source_root = 5;
+  string default_branch = 6;
+}
+
+message DeriveCodeMappingsRequest {
+  int64 organization_id = 1;
+  int64 project_id = 2;
+  repeated string stacktrace_paths = 3;
+}
+
+message DeriveCodeMappingsResponse {
+  repeated CodeMapping mappings = 1;
+}
+
+message CreateExternalIssueRequest {
+  int64 organization_id = 1;
+  int64 integration_id = 2;
+  string title = 3;
+  string description = 4;
+  map<string, string> metadata = 5;
+}
+
+message LinkExternalIssueRequest {
+  int64 group_id = 1;
+  int64 external_issue_id = 2;
+}
```

Comment:
We likely don't want to surface these at all in this service. These are project management integration concerns that happen to be included on one of the SCM base classes.